cmake/modules/ProjectVersioning.cmake

```diff
@@ -0,0 +1,92 @@
+define_property (TARGET
+  PROPERTY GIT_DESCRIBE
+  BRIEF_DOCS "Advanced version info for developers"
+  FULL_DOCS "String with information that is important for developers during
+  development process. This information includes git commit hash, durty status
+  of repo, distance from the last tag.")
+
+define_property (TARGET
+  PROPERTY GIT_UNTRACKED_FILES
+  BRIEF_DOCS "Information about presence of untracked files"
+  FULL_DOCS "Used in helper functions generation to add .with-untracked suffix
+  to version string. Suffix is only added if there are some untracked not
+  ignored files in repository.")
+
+
+function (target_version_information
+	TARGET_NAME i_target_name
+    EXPORT_HEADER i_export_header
+    EXPORT_MACRO i_export_macro
+    VERSIONED_ENTITY i_versioned_entity)
+  find_file (
+	headerFileTemplate
+	"ProjectVersioning/version.h.in"
+	PATHS ${CMAKE_MODULE_PATH})
+
+  find_file (
+	sourceFileTemplate
+	"ProjectVersioning/version.c.in"
+	PATHS ${CMAKE_MODULE_PATH})
+
+  # Общие функции, выполняемые для версионирования проекта.
+
+  # Информация о коммите.
+  #
+  # Необходима для идентицикации версий на этапе разработки между релизами. В
+  # документации для внешнего заказчика может обозначаться, как служебная
+  # информация о релизе.
```

Comment:
is this file taken from somewhere else? why are there comments in Cyrillic?

Reply:
Yep, my fault. It was written some time ago for some local projects that are not supposed to be developed in international community. I shall remove this.